### PR TITLE
improve-col-model-tests-and-fix-ide-and-agents-rules

### PR DESCRIPTION
**Pull request by SalvatorePreviti**
Before raising the next column model rewrite pull request, we need to increase test coverage, so this add more test code.
Add ".skip" in the edge cases / bugs i found in latest and will be fixed by my next PR.

Since new versions of vscode plugins were released and ts version, we are missing some functionality in the ide:
    - update .vscode/settings.json to remove the deprecated or removed settings that were not picked up, this resulted in eslint rules not showing up correctly in the ide like lonely if or no ternary
    - update enterprise tsconfig to resolve to source files, or else it tries to resolve .d.ts in dist, so "find all references" and "got to definition" do not work and go to the built .d.ts files

See https://raw.githubusercontent.com/microsoft/vscode/main/extensions/typescript-language-features/package.json for the list of keys supported in settings.json
 
This also updates agents instructions to follow our code style and testing style, this covers most of the mistakes I usually see in the agents writing code, this should reduce wasted refactoring loops and wasted lint rules misses.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -24,9 +24,10 @@
     "**/node_modules": true,
     "**/typings": true
   },
-  "typescript.preferences.autoImportFileExcludePatterns": ["**/main", "**/main-umd*"],
-  "typescript.tsdk": "node_modules/typescript/lib",
-  "eslint.experimental.useFlatConfig": true,
+  "js/ts.preferences.autoImportFileExcludePatterns": ["**/main", "**/main-umd*"],
```
[the remark this answers is not in the document]
Comment (SalvatorePreviti) [reply .vscode/settings.json:27]: the new versions of the plugin mark the old keys as deprecated

File: .vscode/settings.json [hunk 2]
```diff
@@ -24,9 +24,10 @@
...
+  "js/ts.tsdk.path": "node_modules/typescript/lib",
```
[the remark this answers is not in the document]
Comment (SalvatorePreviti) [reply .vscode/settings.json:28]: the new versions of the plugin mark the old keys as deprecated

File: .vscode/settings.json [hunk 3]
```diff
@@ -109,7 +110,7 @@
   ],
   "git.autoRepositoryDetection": "subFolders",
   "git.repositoryScanIgnoredFolders": ["node_modules", "dist", "typings"],
-  "typescript.enablePromptUseWorkspaceTsdk": true,
+  "js/ts.tsdk.promptToUseWorkspaceVersion": true,
```
[the remark this answers is not in the document]
Comment (SalvatorePreviti) [reply .vscode/settings.json:113]: the new versions of the plugin mark the old keys as deprecated

**Review by StephenCooper — APPROVED** [review]
Thanks for resolving the vsCode and linting issues!
